prowler/providers/azure/services/appinsights/appinsights_service.py

```diff
@@ -45,3 +48,4 @@ class Component:
     resource_id: str
     resource_name: str
     location: str
+    instrumentation_key: str
```

Comment:
This attribute cannot store a None value if `instrumentation_key` does not exist.